src/VariationPriceHandler.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Drupal\commerce_attributes;
+
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\commerce_price\Price;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\commerce_product\ProductAttributeFieldManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\commerce_attributes\Entity\ProductAttributeValue;
+use Drupal\commerce_attributes\Entity\ProductAttribute;
+
+/**
+ * Handles the Variation price calculations.
+ */
+class VariationPriceHandler {
+
+  /**
+   * Logger Service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $Logger;
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
```

Comment:
**CHANGE REQUEST**

You don't need the entity manager; you already have the entity type manager to get the storages from. The entity manager simply calls the entity type manager internally in these cases for backwards compatibility reasons.